src/dataset-operator/pkg/admissioncontroller/mutate.go

```diff
@@ -127,53 +132,49 @@ func Mutate(body []byte) ([]byte, error) {
 			}
 
 			var values []interface{}
-			// if there are no configs to inject, then do no harm.
-			if len(configs_toinject) != 0 {
-				for _, config_toinject := range configs_toinject {
-					//TODO: Check if the configmap reference exists in the API server
-
-					configmap_ref := map[string]interface{}{
-						"prefix": config_toinject,
-						"configMapRef": map[string]interface{}{
-							"name": config_toinject,
-						},
-					}
-					// We also have to inject the companion secret. We are using the convention followed
-					// in the controller where the names of the configmap and the secret are the same.
-					secret_ref := map[string]interface{}{
-						"prefix": config_toinject,
-						"secretRef": map[string]interface{}{
-							"name": config_toinject,
-						},
-					}
-
-					values = append(values, configmap_ref)
-					values = append(values, secret_ref)
+			for _, config_toinject := range configs_toinject {
+				//TODO: Check if the configmap reference exists in the API server
+
+				configmap_ref := map[string]interface{}{
+					"prefix": config_toinject,
```

Comment:
maybe add a "-" after the prefix otherwise env variables names look weird such as **example-datasetbucket**. Please do the same with the secret